mithril-client-cli/src/commands/cardano_db/download.rs

```diff
@@ -60,23 +60,51 @@ pub struct CardanoDbDownloadCommand {
 }
 
 impl CardanoDbDownloadCommand {
-    /// Is JSON output enabled
-    pub fn is_json_output_enabled(&self) -> bool {
-        self.shared_args.json
-    }
-
     /// Command execution
     pub async fn execute(&self, context: CommandContext) -> MithrilResult<()> {
+        let params: ConfigParameters = context.config_parameters()?.add_source(self)?;
+        let prepared_command: PreparedCardanoDbDownload = self.prepare(&params)?;
+
+        prepared_command.execute(context.logger(), params).await
+    }
+
+    fn prepare(&self, params: &ConfigParameters) -> MithrilResult<PreparedCardanoDbDownload> {
+        let mut ancillary_verification_key = None;
         if self.include_ancillary {
             AncillaryLogMessage::warn_ancillary_not_signed_by_mithril();
+            ancillary_verification_key = Some(params.require("ancillary_verification_key")?);
         } else {
             AncillaryLogMessage::warn_fast_bootstrap_not_available();
         }
+        let download_dir = params.require("download_dir")?;
+
+        Ok(PreparedCardanoDbDownload {
+            shared_args: self.shared_args.clone(),
+            digest: self.digest.clone(),
+            download_dir,
```

Comment:
I don't think it's worth creating a variable for that.
```suggestion
            params.require("download_dir")?,
```